iroh-net/src/endpoint.rs

```diff
@@ -334,50 +379,197 @@ impl Endpoint {
+    }
+
+    /// Returns the [`RelayUrl`] of the Relay server used as home relay.
     ///
-    /// Upon the first creation, the first local endpoint discovery might still be underway, in
-    /// this case the first item of the stream will not be immediately available.  Once this first
-    /// set of local endpoints are discovered the stream will always return the first set of
-    /// endpoints immediately, which are the most recently discovered endpoints.
+    /// Every endpoint has a home Relay server which it chooses as the server with the
+    /// lowest latency out of the configured servers provided by [`Builder::relay_mode`].
+    /// This is the server other iroh-net nodes can use to reliably establish a connection
+    /// to this node.
     ///
-    /// The list of endpoints yielded contains both the locally-bound addresses and the
-    /// endpoint's publicly-reachable addresses, if they could be discovered through STUN or
-    /// port mapping.
+    /// Returns `None` if we are not connected to any Relay server.
+    ///
+    /// Note that this will be `None` right after the [`Endpoint`] is created since it takes
+    /// some time to connect to find and connect to the home relay server.  Use
+    /// [`Endpoint::watch_home_relay`] to wait until the home relay server is available.
+    pub fn my_relay(&self) -> Option<RelayUrl> {
+        self.msock.my_relay()
+    }
+
+    /// Watches for changes to the home relay.
+    ///
+    /// If there is currently a home relay it will be yielded immediately as the first item
+    /// in the stream.  This makes it possible to use this function to wait for the initial
+    /// home relay to be known.
+    ///
+    /// Note that it is not guaranteed that a home relay will ever become available.  If no
+    /// servers are configured with [`Builder::relay_mode`] this stream will never yield an
+    /// item.
+    pub fn watch_home_relay(&self) -> impl Stream<Item = RelayUrl> {
+        self.msock.watch_home_relay()
+    }
+
+    /// Returns the local IP endpoints in use as a stream.
+    ///
+    /// The [`Endpoint`] continuously monitors the local IP endpoints, the IP addresses it
+    /// can be reached on by other nodes, for changes.  Whenever changes are detected this
+    /// stream will yield a new list of endpoints.
+    ///
+    /// Upon the first creation, the first local IP endpoint discovery might still be
+    /// underway, in this case the first item of the stream will not be immediately
+    /// available.  Once this first set of local IP endpoints are discovered the stream will
+    /// always return the first set of IP endpoints immediately, which are the most recently
+    /// discovered IP endpoints.
+    ///
+    /// The list of IP endpoints yielded contains both the locally-bound addresses and the
+    /// [`Endpoint`]'s publicly-reachable addresses, if they could be discovered through
+    /// STUN or port mapping.
```

Comment:
Waiting on others to comment in internal discussions, but the `Ip Endpoint` term is not working for me here. I don't find it clear from the naming, it has never been introduced before and from our discussions, it's also slightly inaccurate. 

One could say it's being introduced here in 
> the local IP endpoints, the IP addresses it can be reached on by other nodes

and yet it does not feel clear to me, in particular since `direct addresses` has been used as the remote counterpart. With this I mean: the ip-port pairs we use to connect to other nodes, we have consistently called `direct addresses`, yet the ip-port pairs that can be used to connect to us have a different name. From an user perspective, I don't think this provides really any value and instead generates confussion. It would be more useful to just use the same for both imo.

Reply:
Yeah, I agree this isn't very satisfactory yet.  In this PR I don't want to change the function name yet, but I think that's how it will turn out.  Just don't yet know what to.  But we should get the doc comment working into a better shape already.

Reply:
ok, doubling down on "direct addresses" for this term.  not changing APIs in this PR though, leaving that cleanup as a followup.